operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java

```diff
@@ -160,6 +162,22 @@ protected List<V1EnvVar> createEnv(AuxiliaryImage auxiliaryImage, String name) {
     return vars;
   }
 
+  protected V1ResourceRequirements createResources() {
+    V1ResourceRequirements resources = getServerSpec().getResources();
+    V1ResourceRequirements resourceRequirements = null;
+    if (!resources.getLimits().isEmpty()) {
```

Comment:
Can V1ResourceReqirements::getLimits() return null and cause an NPE here

Reply:
I don't this so. We, by default, initialize the resource requests and limits with empty hash maps. 
```
  private final V1ResourceRequirements resources =
      new V1ResourceRequirements().limits(new HashMap<>()).requests(new HashMap<>());
```